modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/SabrParametersSwaptionVolatilities.java

```diff
@@ -0,0 +1,630 @@
+  private final FixedIborSwapConvention convention;
+  /** 
+   * The valuation date-time. 
+   * <p>
+   * All data items in this environment are calibrated for this date-time. 
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final ZonedDateTime valuationDateTime;
+  /** 
+   * The day count applicable to the model. 
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final DayCount dayCount;
+
+  //-------------------------------------------------------------------------
+  /**
+   * Obtains an instance from the SABR model parameters and the date-time for which it is valid.
+   * 
+   * @param parameters  the SABR model parameters
+   * @param convention  the swap convention for which the data is valid
+   * @param valuationDateTime  the valuation date-time
+   * @param dayCount  the day count applicable to the model
+   * @return the provider
+   */
+  public static SabrParametersSwaptionVolatilities of(
+      SabrInterestRateParameters parameters,
+      FixedIborSwapConvention convention,
+      ZonedDateTime valuationDateTime,
+      DayCount dayCount) {
+
+    return new SabrParametersSwaptionVolatilities(parameters, convention, valuationDateTime, dayCount);
+  }
+
+  /**
+   * Obtains an instance from the SABR model parameters and the date, time and zone for which it is valid.
+   * 
+   * @param parameters  the SABR model parameters
+   * @param convention  the swap convention for which the data is valid
+   * @param valuationDate  the valuation date
+   * @param valuationTime  the valuation time
+   * @param valuationZone  the valuation time zone
+   * @param dayCount  the day count applicable to the model
+   * @return the provider
+   */
+  public static SabrParametersSwaptionVolatilities of(
+      SabrInterestRateParameters parameters,
+      FixedIborSwapConvention convention,
+      LocalDate valuationDate,
+      LocalTime valuationTime,
+      ZoneId valuationZone,
+      DayCount dayCount) {
+
+    return of(parameters, convention, valuationDate.atTime(valuationTime).atZone(valuationZone), dayCount);
+  }
+
+  //-------------------------------------------------------------------------
+  @Override
+  public double volatility(double expiry, double tenor, double strike, double forwardRate) {
+    return parameters.volatility(expiry, tenor, strike, forwardRate);
+  }
+
+  @Override
+  public ValueDerivatives volatilityAdjoint(double expiry, double tenor, double strike, double forward) {
+    return parameters.volatilityAdjoint(expiry, tenor, strike, forward);
+  }
+
+  @Override
+  public double shift(double expiry, double tenor) {
+    return parameters.shift(expiry, tenor);
+  }
+
+  @Override
+  public SurfaceCurrencyParameterSensitivity surfaceCurrencyParameterSensitivity(SwaptionSensitivity pointSensitivity) {
+    throw new UnsupportedOperationException("Sensitivity is based on SwaptionSabrSensitivity, not SwaptionSensitivity");
```

Comment:
This design is the best we can do at the moment. It requires another abstraction for point sensitivities, which we don't have time for right now, but which should be possible. It is on my list.